src/systemctl/systemctl-edit.c

```diff
@@ -400,6 +460,42 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
         return 0;
 }
 
+static int trim_edit_markers(const char *path) {
+        _cleanup_free_ char *contents = NULL;
+        char *contents_start = NULL;
+        const char *contents_end = NULL;
+        size_t size;
+        int r;
+
+        /* Trim out the lines between the two markers */
+        r = read_full_file(path, &contents, &size);
+        if (r < 0) {
+                log_error_errno(r, "Failed to read temporary file \"%s\": %m", path);
+                return r;
+        }
```

Comment:
log_error_errno() returns the errror you pass in, hence plese shorten these four lines to:

```c
if (r < 0)
        return log_error_errno(r, …);
```